token.go

```diff
@@ -214,16 +214,61 @@ type TSTInfo struct {
 	Extensions     []pkix.Extension `asn1:"optional,tag:1"`
 }
 
+// Timestamp denots the time at which the timestamp token was created by the TSA
+//
+// Reference: RFC 3161 2.4.2
+type Timestamp struct {
+	// Value is the GenTime of TSTInfo
+	Value time.Time
+
+	// Accuracy is the Accuracy of TSTInfo
+	Accuracy time.Duration
+}
+
+// Before returns true if the lower limit of the time at which the timestamp
+// token was created is before u.
+//
+// Reference: RFC 3161 2.4.2
+func (t *Timestamp) Before(u time.Time) bool {
+	timestampLowerLimit := t.Value.Add(-t.Accuracy)
+	return timestampLowerLimit.Before(u)
+}
+
+// After returns true if the upper limit of the time at which the timestamp
+// token was created is after u.
+//
+// Reference: RFC 3161 2.4.2
+func (t *Timestamp) After(u time.Time) bool {
+	timestampUpperLimit := t.Value.Add(t.Accuracy)
```

Comment:
Mathematically, should it be
```suggestion
	timestampLowerLimit := t.Value.Add(-t.Accuracy)
```

`[2:29, 2:31]` is after `2:28` but not after `2:30`.